src/tests/zf_apps/zfsink.c

```diff
@@ -233,8 +247,51 @@ void print_attrs(struct zf_attr* attr)
 }
 
 
+static void monitor(struct resources* res)
+{
+  uint64_t now_bytes, prev_bytes;
+  struct timeval start, end;
+  uint64_t prev_pkts, now_pkts;
+  int ms, pkt_rate, mbps;
+
+  printf("#%9s %16s %16s", "pkt-rate", "bandwidth(Mbps)", "total-pkts\n");
+
+  prev_pkts = res->n_rx_pkts;
+  prev_bytes = res->n_rx_bytes;
+  gettimeofday(&start, NULL);
+
+  while( 1 ) {
+    sleep(1);
+    now_pkts = res->n_rx_pkts;
+    now_bytes = res->n_rx_bytes;
+    gettimeofday(&end, NULL);
+    ms = (end.tv_sec - start.tv_sec) * 1000;
+    ms += (end.tv_usec - start.tv_usec) / 1000;
+    pkt_rate = (int) ((now_pkts - prev_pkts) * 1000 / ms);
+    mbps = (int) ((now_bytes - prev_bytes) * 8 / 1000 / ms);
+    pthread_mutex_lock(&printf_mutex);
```

Comment:
I don't think a mutex is needed for printf. I believe it synchronises itself at least in my experiments.